pennylane/data/dataset.py

```diff
@@ -186,9 +188,19 @@ def read(self, filepath, lazy=False):
         if attribute == "full":
             data = self._read_file(filepath)
             for attr, value in data.items():
-                setattr(self, f"{attr}", None if lazy else value)
```

Comment:
I'm going to remove `lazy` from this in another PR, it doesn't quite make sense given regular use cases. But for now, I have this whole mess going on